checkers/time.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package checkers
+
+import (
+	"fmt"
+	"math"
+	"reflect"
+	"time"
+
+	gc "gopkg.in/check.v1"
+)
+
+type timeCompareChecker struct {
+	*gc.CheckerInfo
+	compareFunc func(time.Time, time.Time) bool
+}
+
+// After checks whether the obtained time.Time is After the want time.Time.
+var After gc.Checker = &timeCompareChecker{
+	CheckerInfo: &gc.CheckerInfo{Name: "After", Params: []string{"obtained", "want"}},
+	compareFunc: func(t1, t2 time.Time) bool {
+		return t1.After(t2)
+	},
+}
+
+// Before checks whether the obtained time.Time is Before the want time.Time.
+var Before gc.Checker = &timeCompareChecker{
+	CheckerInfo: &gc.CheckerInfo{Name: "Before", Params: []string{"obtained", "want"}},
+	compareFunc: func(t1, t2 time.Time) bool {
+		return t1.Before(t2)
+	},
+}
+
+// Almost checks whether the obtained time.Time is within 1s of the want time.Time.
+var Almost gc.Checker = &timeCompareChecker{
```

Comment:
Can we add an option param for the time window? Default to 1s.

Reply:
can just use After and Before in combination.